src/m365/planner/commands/roster/roster-user-plan-list.ts

```diff
@@ -0,0 +1,101 @@
+import auth from '../../../../Auth';
+import { Logger } from '../../../../cli/Logger';
+import GraphCommand from '../../../base/GraphCommand';
+import GlobalOptions from '../../../../GlobalOptions';
+import commands from '../../commands';
+import { validation } from '../../../../utils/validation';
+import { accessToken } from '../../../../utils/accessToken';
+import { odata } from '../../../../utils/odata';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId?: string;
+  userName?: string;
+}
+
+class PlannerRosterUserPlanListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.ROSTER_USER_PLAN_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all Roster plans for a specified user';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
```

Comment:
Let's also use option sets for the two variables. The option set should have a runsWith so it only validates when one of the two options is provided.